framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1014,6 +1188,47 @@ private AnnotationMirror createNumberAnnotationMirror(List<Number> values) {
     }
 
     /**
+     * Create an {@code @IntRange} annotation from the two (inclusive) bounds. Does not return
+     * BOTTOMVAL or UNKNOWNVAL.
+     */
+    public AnnotationMirror createIntRangeAnnotation(long from, long to) {
+        assert from <= to;
+        AnnotationBuilder builder = new AnnotationBuilder(processingEnv, IntRange.class);
+        builder.setValue("from", from);
+        builder.setValue("to", to);
+        return builder.build();
+    }
+
+    /**
+     * Create an {@code @IntRange} annotation from the range. May return BOTTOMVAL or UNKNOWNVAL.
+     */
+    public AnnotationMirror createIntRangeAnnotation(Range range) {
+        if (range.isNothing()) {
+            return BOTTOMVAL;
+        } else if (range.isEverything()) {
+            return UNKNOWNVAL;
+        } else {
+            return createIntRangeAnnotation(range.from, range.to);
+        }
+    }
+
+    /** Converts an {@code @IntVal} annotation to an {@code @IntRange} annotation. */
+    public AnnotationMirror convertIntValToIntRange(AnnotationMirror intValAnno) {
+        List<Long> intValues = getIntValues(intValAnno);
+        return createIntRangeAnnotation(Collections.min(intValues), Collections.max(intValues));
+    }
+
+    public static Range getIntRange(AnnotationMirror rangeAnno) {
```

Comment:
Some short javadoc like for the other methods would be nice. Especially for AnnotationMirror parameters I find it nice to see what the expected annotation is.